app/main/system/serializers.ts

```diff
@@ -0,0 +1,19 @@
+import * as f from "fp-ts/lib/function"
+import { Lifecycle, pureComponent } from "./lifecycle"
+
+export type Serializer = {
+  encode(value: any): Promise<string>;
+  decode(text: string): Promise<any>;
+}
+
```

Comment:
`Serializer` can be generalized as:
```ts
export type Serializer<A, B> = {
  encode(value: A): Promise<B>
  decode(serialized: B): Promise<a>
}
```

Then `json` would be a `Lifecycle<Serializer<JsonSerializable, string>, void>` where:
```ts
export type JsonSerializable = number | string | boolean | JsonSerializableObject | JsonSerializableArray
export interface JsonSerializableArray extends Array<JsonSerializable> { }
export interface JsonSerializableObject {[key: string]: JsonSerializable}
```